react/typescript/vis-and-filter-components/src/__mocks__/render_with_extension.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2022 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import type { ReactNode } from 'react'
+import React from 'react'
+import { renderWithTheme } from '@looker/components-test-utils'
+import type { RenderResult } from '@testing-library/react'
+import type { ExtensionContextData } from '@looker/extension-sdk-react'
+import { ExtensionContext } from '@looker/extension-sdk-react'
+import type { ExtensionHostApi, ExtensionSDK } from '@looker/extension-sdk'
+import { registerHostApi } from '@looker/extension-sdk'
+import { MemoryRouter } from 'react-router-dom'
+
+const getExtensionSDK = (extensionSDKOverride: Partial<ExtensionSDK>) => {
+  const extensionSDK = {
+    lookerHostData: {},
+    error: () => {
+      // noop
+    },
+    ...extensionSDKOverride,
+  } as ExtensionHostApi
+  registerHostApi(extensionSDK)
+  return extensionSDK
+}
+
+const getExtensionContext = (
+  extensionSDKOverride: Partial<ExtensionSDK>,
+  contextOverride: Partial<ExtensionContextData>
+): ExtensionContextData =>
+  ({
+    extensionSDK: getExtensionSDK(extensionSDKOverride),
+    coreSDK: {},
+    route: '',
+    ...contextOverride,
+  } as ExtensionContextData)
+
+const withExtensionContext = (
+  component: ReactNode,
+  extensionSDKOverride: Partial<ExtensionSDK>,
+  contextOverride: Partial<ExtensionContextData>
+) => (
+  <MemoryRouter>
+    <ExtensionContext.Provider
+      value={getExtensionContext(extensionSDKOverride, contextOverride)}
+    >
+      {component}
+    </ExtensionContext.Provider>
+  </MemoryRouter>
+)
+
+export const renderWithExtensionContext = (
```

Comment:
I just copied these over from the `helloworld-ts` directory. Is it worth moving them over to a shared mocks directory somewhere?

Reply:
I've duplicated them because I want a developer to be able to just copy an example directory and have it just work. Me thinks there will be some friction if we have some shared mocks. If you can think of a better approach I'd be happy to use it.

Reply:
Oh that's a good point! Yeah keeping each example self sufficient is a good call. I wonder if it would make sense to publish them in the [components test utils](https://github.com/looker-open-source/components/tree/main/packages/components-test-utils)? Or maybe spin up some similar such package focused on extension framework test utils?

Reply:
That makes sense to me.